tools/downloader/topologies.py

```diff
@@ -0,0 +1,233 @@
+        continue
+
+    _module = types.ModuleType(_moduleName)
+    sys.modules['topologies.' + _moduleName] = _module
+    globals()[_moduleName] = _module
+    _module.__dict__['sys'] = sys
+    _module.__dict__['os'] = os
+
+    # List of models which have versions (mostly DLDT models). For such kind of models
+    # with the highest version we will add short names. In example, for
+    # "face_detection_retail_0004" and "face_detection_retail_0005" method with name
+    # "face_detection_retail" returns result of "face_detection_retail_0005".
+    _versionedNames = {}
+
+    for _topologyName in os.listdir(_modelsSubDir):
+        _config = os.path.join(_modelsSubDir, _topologyName, 'model.yml')
+        if not os.path.exists(_config):
+            continue
+
+        with open(_config, 'rt') as _f:
+            _topology = yaml.safe_load(_f)
+            _name = _topologyName.replace('-', '_').replace('.', '_')
+            _description = _topology['description']
+            _license = _topology['license']
+            _files = _topology['files']
+            _framework = _topology['framework']
+            _taskType = _topology['task_type']
+            _configPath = ''
+            _modelPath = ''
+            _downloadDir = os.path.join(_cache, _moduleName, _topologyName)
+            _moArgs = {}
+            if 'model_optimizer_args' in _topology:
+                for _arg in _topology['model_optimizer_args']:
+                    _tokens = _arg.split('=')
+                    _moArgs[_tokens[0]] = _tokens[1] if len(_tokens) == 2 else None
+
+
+            assert(len(_files) > 0)
+            if len(_files) > 2:
+                assert(_framework == 'dldt'), ('Unexpected framework type: ' + _framework)
+                # Get only basename and add precision marker
+                _configPath = _files[0]['name']
+                _modelPath = _files[1]['name']
+
+                _pos = _configPath.find('/')
+                if _configPath[:_pos] in common.KNOWN_PRECISIONS:
+                    _configPath = _precisionMarker + _configPath[_pos:]
+
+                _pos = _modelPath.find('/')
+                if _modelPath[:_pos] in common.KNOWN_PRECISIONS:
+                    _modelPath = _precisionMarker + _modelPath[_pos:]
+
+            else:
+                _configPath = _files[0]['name']
+                _modelPath = _files[-1]['name']
+
+            # To manage origin files location from archives
+            if '--input_model' in _moArgs:
+                _modelPath = _moArgs['--input_model'].replace('$dl_dir/', '')
+            if '--input_proto' in _moArgs:
+                _configPath = _moArgs['--input_proto'].replace('$dl_dir/', '')
+
+            if _configPath:
+                _configPath = os.path.realpath(os.path.join(_downloadDir, _configPath))
+            if _modelPath:
+                _modelPath = os.path.realpath(os.path.join(_downloadDir, _modelPath))
+
+
+            exec("""
+class {className}:
```

Comment:
Does each model really need to be its own class? I'd think that they could all be instances of the same class.

Reply:
That's a good point. I'll check how docstring will look like in this way.

Reply:
Created a base class. Thanks!